### ISSUE TITLE
Paniced at `hot_targets.gete(t).unwrap()`

### ISSUE THREAD

**Issue by honhimW** [gin66/tui-logger#69]
In the `tui-logger` crate version 0.12.0, there is a panic occurring at `src/lib.rs:833:59` with the following code:

```rust
let t = &self.targets[i + offset];
let hot_level_filter = hot_targets.get(t).unwrap(); // Paniced!
let level_filter = targets.get(t).unwrap();
```
https://github.com/gin66/tui-logger/blob/64129b13dc327a00342e110f81ce85c16da14656/src/lib.rs#L842-L846
> tui-logger-0.12.0\src\lib.rs:833:59:
> called `Option::unwrap()` on a `None` value

> The panic happens occasionally under certain conditions, but the exact cause is unclear.

**Comment by gin66** [comment]
Thanks for the report. I have modified the code and hope, that it is now fixed.

**Comment by gin66** [comment]
Remaining issue is, that there is still an unexplained root cause. So the fix only masks an underlying bug….

**Comment by honhimW** [comment]
My current usage involves first creating a `TuiWidgetState` and then setting the level for a specific target. However, if the target does not have any logs yet, this error occurs. Is this by design?

In my current setup, I set the default output level to info and also set default levels for certain targets.
```rust
let mut state = TuiWidgetState::new().set_default_display_level(LevelFilter::Info);
state = state.set_level_for_target("some::logger", LevelFilter::Off);
```

Additionally, another question is that most of logging systems are typically asynchronous(I have not fully read your code yet, sorry~). This means that even if I proactively log a message before call the render method, this error might still occur, right?.

**Comment by gin66** [comment]
Thanks for the explanation and based on your input I have been able to reproduce the problem. Based on my investigation, there has not been an underlying root cause. It is just an oversight, that every list can have an own mapping of target to log filter value, which is not mirrored to the logger. Consequently, the `unwrap()` is incorrect and `unwrap_or()` is a valid fix. I have added a comment to explain this in the code.

The new version is already published.
